File: BackEnd/ExamManagement/DPD.HR.Business.Dto/Dto/WorkflowStepDto.cs
namespace DPD.HumanResources.Dtos.Dto;

public class WorkflowStepDto
{
    public Guid Id {get;set;}
    public Guid TemplateId {get;set;}
    public string? StepName {get;set;}
    public Guid IdRoleAssign {get;set;}
    public Guid IdUnitAssign {get;set;}
    public bool AllowTerminated {get;set;}
    public string? RejectName {get;set;}
    public int Order {get;set;}
    public string? OutCome {get;set;}
    public DateTime? CreatedDate {get;set;}
    public int? Status {get;set;}
    public string? ProcessWorkflowButton {get;set;}
    public string? StatusColor {get;set;}
    public bool IsDirectUnit {get;set;}
}

File: FrontEnd/src/models/workflowStepDTO.ts
export interface WorkflowStepDTO {
  id: string;
  templateId: string;
  stepName: string | null;
  idRoleAssign: string;
  idUnitAssign: string | null;
  allowTerminated: boolean;
  rejectName: string | null;
  order: number;
  outCome: string | null;
  createdDate: Date;
  status: number | null;
  processWorkflowButton: string | null;
  statusColor: string | undefined;
  isDirectUnit: boolean;
}
